main.go

```diff
@@ -56,6 +59,18 @@ func main() {
 			"last": func(x int, a interface{}) bool {
 				return x == reflect.ValueOf(a).Len()-1
 			},
+			"loop": func(n, m int) []int {
```

Comment:
This is fine for now, but we will need to move these functions into their own "template-helper" package otherwise this file will get too long. Fine for now, but worth remembering!